### PR TITLE
Improve IERS docs and add a top-level entry

### PR DESCRIPTION
**Pull request by taldcroft**
### Description

This pull request is to provide more background on the IERS-A and -B files. It also adds a top-level entry for the `utils.iers` sub-package. Questions about this sub-package seem to come up a lot and I always find myself needing to google it to find the docs.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [x] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [x] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label.
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

File: docs/utils/iers.rst [hunk 1]
```diff
@@ -7,23 +7,40 @@ IERS data access (`astropy.utils.iers`)
...
+times.  The UT1-UTC values are used in :ref:`astropy-time` to provide UT1 values, and
+the polar motions are used in `astropy.coordinates` to determine Earth
+orientation for celestial-to-terrestrial coordinate transformations.
 
 .. note:: The package also provides machinery to track leap seconds.  Since it
           generally should not be necessary to deal with those by hand, this
           is not discussed below.  For details, see the documentation of
           `~astropy.utils.iers.LeapSeconds`.
 
+There are two IERS data products that are commonly used in the :ref:`astropy-time`
```
Comment (mhvk) [inline docs/utils/iers.rst:24]: They are used both in `time` and `coordinates`; maybe just remove the reference here?

File: docs/utils/iers.rst [hunk 2]
```diff
@@ -7,23 +7,40 @@ IERS data access (`astropy.utils.iers`)
...
+package:
+
+- **Bulletin A** (:class:`~astropy.utils.iers.IERS_A`) is updated weekly and has
+  historical data starting from 1972 and predictive data for 1 year into the
```
Comment (mhvk) [inline docs/utils/iers.rst:28]: 1973

File: docs/utils/iers.rst [hunk 3]
```diff
@@ -54,7 +71,7 @@ Note that you can forcibly clear the download cache as follows::
   >>> from astropy.utils.data import clear_download_cache
   >>> clear_download_cache()  # doctest: +SKIP
 
-The default IERS data used automatically is updated by the service every 7 days
+The default IERS-A data used automatically is updated by the service every 7 days
```
Comment (mhvk) [inline docs/utils/iers.rst:74]: Hah, good reminder to update it when we fix `IERS_Auto`!

File: docs/index.rst [hunk 4]
```diff
@@ -103,6 +103,7 @@ Computations and utilities
 
    cosmology/index
    convolution/index
+   utils/iers
```
Comment (mhvk) [inline docs/index.rst:106]: Should this be in the "Nuts and bolts" section instead?

**Review by mhvk — APPROVED** [review]
Yes, good idea! Some small comments/corrections. Am approving since I'm a bit in and out of contact...

File: docs/index.rst (hunk 4 above)
Comment (taldcroft) [reply docs/index.rst:106]: I thought about that, but I think of Nuts and bolts as pure infrastructure like logging and downloading files. This feels more like science and in fact Earth rotation is a science discipline on par with cosmology.

Comment (mhvk) [reply docs/index.rst:106]: OK, fair enough!

**Pull request merged**